src/output-json.c

```diff
@@ -438,8 +444,69 @@ static void EveAddMetadata(const Packet *p, const Flow *f, JsonBuilder *js)
 
 int CreateJSONEther(JsonBuilder *parent, const Packet *p, const MacSet *ms);
 
+/* Callback function to pack payload contents from a stream into a buffer
+ * so we can report them in JSON output. */
+static int JsonDumpStreamSegmentCallback(const Packet *p, void *data, const uint8_t *buf, uint32_t buflen)
+{
+    MemBuffer *payload = (MemBuffer *)data;
+    MemBufferWriteRaw(payload, buf, buflen);
+
+    return 1;
+}
+
+
+static void EveAddStreamDataOneWay(const Packet *p, const Flow *f, JsonBuilder *js,
+                                    uint8_t flags, MemBuffer *payload, uint8_t flag)
+{
+    StreamSegmentForEach((const Packet *)p, flag,
+            JsonDumpStreamSegmentCallback,
+            (void *)payload);
+    if (payload->offset) {
+        if (flags & JSONLOG_JSON_PAYLOAD_BASE64) {
+            unsigned long len = payload->offset * 2 + 1;
+            uint8_t encoded[len];
+            Base64Encode(payload->buffer, payload->offset, encoded, &len);
```

Comment:
needs to be error checked

Reply:
btw @jasonish I still think a jb_set_string_as_base64 would be useful, also to reduce stack usage.